File: test/features/step_definitions/signInSteps.js
// Step definitions
/* global browser expect */

const {defineSupportCode} = require('cucumber')
defineSupportCode(function ({Given, When, Then}) {
  Given(/^I am on the sign in page$/, function (callback) {
    browser.get(this.config.appUrl)
    expect(this.signIn.usernameField).to.eventually.be.present
      .then(callback)
  })

  Then(/^I should see a page heading$/, function (callback) {
    expect(this.signIn.heading).to.eventually.have.text('Check Development Service')
      .then(callback)
  })

  Then(/^I should see instructions$/, function (callback) {
    expect(this.signIn.instructions).to.eventually.have.text('Sign-in to access the Check Development Service')
      .then(callback)
  })

  Given(/^I have entered valid credentials$/, function (callback) {
    browser.get(this.config.appUrl)
    this.signIn.login('teacher1', 'password')
    callback()
  })

  When(/^I sign in$/, function (callback) {
    this.signIn.signIn.click()
    callback()
  })

  Then(/^I should be taken to the school landing page$/, function (callback) {
    expect(this.landingPage.breadcrumb).to.eventually.have.text('School Home')
      .then(callback)
  })

  Given(/^I have entered invalid credentials$/, function (callback) {
    browser.get(this.config.appUrl)
    this.signIn.login('teacher1', 'wrong')
    callback()
  })

  Then(/^I should be taken to the failed login page$/, function (callback) {
    expect(this.signInFailure.heading).to.eventually.have.text('Unable to confirm details')
    expect(this.signInFailure.instructions).to.eventually.have.text('The details entered do not match our records.')
      .then(callback)
  })
})

File: admin/test/features/step_definitions/sign_in_steps.rb
Given(/^I am on the sign in page$/) do
  sign_in_page.load
end

Then(/^I should see a page heading$/) do
  expect(sign_in_page).to have_heading
end

Then(/^I should see instructions$/) do
  expect(sign_in_page).to have_instructions
end

Given(/^I have entered valid credentials$/) do
  sign_in_page.load
  sign_in_page.enter_credentials('teacher1','password')
end

When(/^I sign in$/) do
  sign_in_page.sign_in.click
end

Given(/^I have entered invalid credentials$/) do
  sign_in_page.load
  sign_in_page.enter_credentials('teacher1', 'wrong')
end

Then(/^I should be taken to the failed login page$/) do
  expect(sign_in_failure_page).to be_displayed
end

Given(/^I am logged in$/) do
  sign_in_page.load
  sign_in_page.login('teacher1', 'password')
end

Then(/^I should be taken to the school landing page$/) do
  expect(school_landing_page).to be_displayed
end

Given(/^I have signed in with (.*)$/) do |teacher|
  @teacher = teacher
  sign_in_page.load
  sign_in_page.login(@teacher,'password')
end

Given(/^I am on the sign in failure page$/) do
  sign_in_failure_page.load
end

When(/^I decide to get in contact$/) do
  sign_in_page.contact.click
end

Then(/^I should be taken to the contact page for mtc$/) do
  expect(contact_page).to be_displayed
end